docs/source/deprecation.md

```diff
@@ -1,14 +1,164 @@
+example simple and say that we do not want to set any replacement of this
+model. The first thing we have to do is to set the model definition as deprecated:
 
-To help with the deprecation process of entities in leapp, the leapp framework introduced a decorator `deprecated`
-which can be imported from `leapp.utils.deprecation`.
+```python
+from leapp.models import Model, fields
+from leapp.topics import SomeTopic
+from leapp.utils.deprecation import deprecated
+
+
+@deprecated(since='2020-06-20', message='This model has been deprecated.')
+class Foo(Model):
+    topic = SomeTopic
+    value = fields.String()
+```
+
+If we do only this and execute actors that produce/consume messages of this model,
+we will obtain messages like these (just example from the actor
+producing the message after execution by snactor):
+
+```bash
+# snactor run fooproducer
+============================================================
+                 USE OF DEPRECATED ENTITIES
+============================================================
+
+Usage of deprecated Model "Foo" @ /path/to/repo/actors/fooproducer/actor.py:17
+Near:         self.produce(Foo(value='Answer is: 42'))
+
+Reason: This model has been deprecated.
+------------------------------------------------------------
+
+============================================================
+             END OF USE OF DEPRECATED ENTITIES
+============================================================
+```
+
+Obviously we do not want to remove the actor yet, because in such a case, the
+model could be hardly called as deprecated. But we do not want the deprecation
+message to be produced in this case, as we want to keep the original
+functionality of that actor during the deprecation protection period.
+The warning messages
+are focused on a "custom" use of the deprecated model - in case a user
+creates their own
+actor producing/consuming a message of the model. To fix this, suppress
+the deprecation message in this actor. To do it, the only thing you have
+to do is to set the `suppress_deprecation` decorator before the actor,
+e.g.:
+```
+#TODO: add example
+```
 
-This decorator will mark items as being deprecated and ensures that a deprecation warning is displayed if this item
-is being used.
+Instead of that, I will set the `suppress_deprecation` for the actor, so
+the messages will not be produced in this case.
 
-### Warning message display behaviour
+## How are users notified about deprecation
 
-Warning messages are always shown after `snactor` executions via `snactor run` or `snactor workflow run`.
+Beside changelogs/release notes, warning messages are always shown after
+`snactor` executions via `snactor run` or `snactor workflow run`.
 At the end of executions of `leapp upgrade` or `leapp preupgrade` the warnings are added as report entries with a
 developer audience. If the deprecation is less than 6 months old the severity is set to 'MEDIUM' if it is older than
 6 months the severity is set to 'HIGH'
```

Comment:
We already facing this third time I guess in the text. Let' have it in one place

Reply:
That's expected as it is draft and the document is not finished.